client/request.go

```diff
@@ -195,44 +185,59 @@ func (cli *Client) doRequest(req *http.Request) (serverResponse, error) {
 			}
 		}
 
-		return serverResp, errConnectionFailed{errors.Wrap(err, "error during connect")}
+		return nil, errConnectionFailed{errors.Wrap(err, "error during connect")}
 	}
 
-	if resp != nil {
-		serverResp.statusCode = resp.StatusCode
-		serverResp.body = resp.Body
-		serverResp.header = resp.Header
-	}
-	return serverResp, nil
+	return resp, nil
 }
 
-func (cli *Client) checkResponseErr(serverResp serverResponse) error {
-	if serverResp.statusCode >= 200 && serverResp.statusCode < 400 {
+func (cli *Client) checkResponseErr(serverResp *http.Response) (retErr error) {
```

Comment:
This function needs a bit of a cleanup; I adjusted it to work with some tests that used non-standard `http.Response` (only partially propagated), but there's a lot of cruft accumulated in this function.